helper/maputil/syncmap.go

```diff
@@ -0,0 +1,83 @@
+package maputil
```

Comment:
If possible, it's nice to avoid `helper` and `util` in package names as they're not very descriptive or prescriptive, could we make this a super tightly scoped package and just call it `syncmap`?

```suggestion
package syncmap
```